addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/discovery/LGWebOSDiscovery.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lgwebos.internal.discovery;
+
+import static org.openhab.binding.lgwebos.LGWebOSBindingConstants.*;
+
+import java.io.File;
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.lgwebos.LGWebOSBindingConstants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.connectsdk.core.Context;
+import com.connectsdk.core.Util;
+import com.connectsdk.device.ConnectableDevice;
+import com.connectsdk.discovery.DiscoveryManager;
+import com.connectsdk.discovery.DiscoveryManagerListener;
+import com.connectsdk.service.command.ServiceCommandError;
+
+/**
+ * This class provides the bridges between openhab thing discovery and connect sdk device discovery.
+ *
+ * @author Sebastian Prehn
+ * @since 1.8.0
+ */
+@Component(service = { DiscoveryService.class, LGWebOSDiscovery.class }, immediate = true, name = "binding.lgwebos")
+public class LGWebOSDiscovery extends AbstractDiscoveryService implements DiscoveryManagerListener, Context {
```

Comment:
LGWebOSDiscovery is for the action to find it. 
I set the names so that it is 1:1 compared to the xml config file.
Aren't the names important to retrieve the config variable "localIP" correctly in LGWebOSDiscovery (binding.lgwebos)?
I probably don't care about the name of LGWebOSAction and of LGWebOSHandlerFactory

Reply:
Indeed, I missed that you reference this service. For the config I think you could use configurationPid argument of the component annotation.

Reply:
Done